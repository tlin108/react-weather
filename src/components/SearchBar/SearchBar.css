.search-location {
  position: relative;
  margin-bottom: 70px; }
  .search-location input[type="text"] {
    width: 100%;
    padding: 20px 50px 20px 20px;
    background: #1e202b;
    color: white; }
  .search-location input[type="submit"] {
    position: absolute;
    top: 5px;
    right: 5px;
    bottom: 5px;
    padding: 0 40px; }