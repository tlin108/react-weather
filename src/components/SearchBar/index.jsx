import React, { Component } from 'react';
import './SearchBar.css';

class SearchBar extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      cityName: ''
    }
  }

  onCityNameChange(cityName) {
    this.setState({ cityName });
  }

  onFormSubmit(e) {
    e.preventDefault();
    const { cityName } = this.state;
    if (cityName === ''){
      return;
    }
    this.props.onSearchSubmit(cityName);
  }

  render() {
    return (
      <form onSubmit={e => this.onFormSubmit(e)} className="search-location">
        <input type="text" placeholder="Search by..." onChange={(e) => this.onCityNameChange(e.target.value)} />
        <button type="submit" value="Search">Submit</button>
      </form>
    );
  }
}

export default SearchBar;