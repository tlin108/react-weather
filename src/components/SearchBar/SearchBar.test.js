import React from 'react';
import { mount, shallow } from 'enzyme';
import SearchBar from './index';

describe('SearchBar test suite', () => {
  const onButtonSubmit = jest.fn();
  const wrapper = shallow(
    <SearchBar onSearchSubmit={onButtonSubmit} />
  );

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('simulates submit event', () => {
    const wrapperMount = mount(
      <SearchBar onSearchSubmit={onButtonSubmit} />
    );
    wrapperMount.find('[type="submit"]').get(0).click();
    expect(onButtonSubmit).toHaveBeenCalled();
  });

  it('simulates onchange event', () => {
    const value = 'Boston';
    expect(wrapper.instance().state.cityName).toEqual('');
    wrapper.find('input').simulate('change', {
      target: { value }
    });
    expect(wrapper.instance().state.cityName).toEqual('Boston');
  });
});
