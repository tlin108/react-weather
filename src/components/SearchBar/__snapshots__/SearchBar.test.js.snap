exports[`SearchBar test suite renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="search-location"
    onSubmit={[Function]}>
    <input
        onChange={[Function]}
        placeholder="Search by..."
        type="text" />
    <button
        type="submit"
        value="Search">
        Submit
    </button>
</form>,
  "nodes": Array [
    <form
      className="search-location"
      onSubmit={[Function]}>
      <input
            onChange={[Function]}
            placeholder="Search by..."
            type="text" />
      <button
            type="submit"
            value="Search">
            Submit
      </button>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchBar
        onSearchSubmit={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchBar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onSearchSubmit": [Function],
        },
        "refs": Object {},
        "state": Object {
          "cityName": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="search-location"
          onSubmit={[Function]}>
          <input
                    onChange={[Function]}
                    placeholder="Search by..."
                    type="text" />
          <button
                    type="submit"
                    value="Search">
                    Submit
          </button>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="search-location"
          onSubmit={[Function]}>
          <input
                    onChange={[Function]}
                    placeholder="Search by..."
                    type="text" />
          <button
                    type="submit"
                    value="Search">
                    Submit
          </button>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBar
    onSearchSubmit={[Function]} />,
}
`;
